### PR TITLE
Implement option to reorder global colors

### PR DESCRIPTION
**Pull request by JeanPaiva**
This PR adds the possibility to reorder the global colors.

### CODE REVIEW

**Comment by tomusborne** [comment]
This is working really nicely. The only issue I'm seeing now is when you add a new color and type in a variable name (that isn't being used), the popover closes and you have to click the circle again.

File: src/customizer-controls/color-manager/GeneratePressColorManagerControlForm.js [hunk 1]
```diff
@@ -3,6 +3,10 @@ import { useCallback, useEffect, useState } from '@wordpress/element';
 import useColors from './hooks/useColors';
 import { isObject, findIndex } from 'lodash';
```
Comment (tomusborne) [inline src/customizer-controls/color-manager/GeneratePressColorManagerControlForm.js:4]: Seeing an eslint error here as we haven't added `lodash` to our deps.

**Comment by SchneiderSam** [comment]
Thanks for this wonderful feature!!!!

**Comment by tomusborne** [comment]
I'm still seeing the following issue when I add a name that's already been taken. It should revert onBlur I think.

https://user-images.githubusercontent.com/20714883/169445320-f64bbfd4-a066-41fe-a3eb-80df5c30ebf7.mp4

**Comment by tomusborne** [comment]
This is working great.

Only thing I've noticed is that #353 isn't fixed due to this line: https://github.com/tomusborne/generatepress/blob/color-manager-improvements/src/customizer-controls/color-manager/GeneratePressColorManagerControlForm.js#L52

The "hard-coded" CSS is added to the footer. So changing it to `document.body` on that line fixes it.

Other than that this is good to go (and a badass PR).

**Pull request merged**